src/components/structures/MatrixChat.js

```diff
@@ -174,7 +176,7 @@ module.exports = React.createClass({
             // Don't auto-register as a guest. This applies if you refresh the page on a
             // logged in client THEN hit the Sign Out button.
             this._autoRegisterAsGuest = false;
-            this.startMatrixClient();
+            Lifecycle.startMatrixClient();
```

Comment:
will this *restart* the client in the case of a `guest_access_token` in the parameters?

(Ultimately I think I'd like to see the restore-client-on-restart code from MatrixClientPeg following a similar codepath to the `guest_access_token`-in-params functionality, since they are both just ways of creating a MatrixClient from known credentials at startup, and would avoid having to expose both `startMatrixClient` and `login` from `Lifecycle`. Probably best left for another PR, though.)